### PR TITLE
JDK-8257514: Fix the issues in jdk.jpackage identified by SpotBugs

### PR DESCRIPTION
**Pull request by andyherrick**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257514](https://bugs.openjdk.java.net/browse/JDK-8257514): Fix the issues in jdk.jpackage identified by SpotBugs

### Reviewers
 * [Alexey Semenyuk](https://openjdk.java.net/census#asemenyuk) (@alexeysemenyukoracle - Committer)
 * [Alexander Matveev](https://openjdk.java.net/census#almatvee) (@sashamatveev - Committer)
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1561/head:pull/1561`
`$ git checkout pull/1561`

### CODE REVIEW

**Comment by andyherrick** [comment]
@sashamatveev  , @azuev-java , @alexeysemenyukoracle please review

**Comment by andyherrick** [comment]
Build failure in above macos debug build is unrelated to this change (CompileDemos.gmk:161)

**Review by alexeysemenyukoracle — APPROVED** [review]

**Comment by alexeysemenyukoracle** [comment]
Honestly, I don't see much value in the new static members of IOUtils class. But they don't make any trouble either.

**Review by sashamatveev — APPROVED** [review]

File: src/jdk.jpackage/share/classes/jdk/jpackage/internal/DeployParams.java [hunk 1]
```diff
@@ -208,7 +208,9 @@ public void validate() throws PackagerException {
                 Arguments.CLIOptions.PREDEFINED_APP_IMAGE.getId());
         if (appImage != null) {
             Path appImageDir = Path.of(appImage);
-            if (!Files.exists(appImageDir) || appImageDir.toFile().list().length == 0) {
+            if (!Files.exists(appImageDir)
+                    || appImageDir.toFile().list() == null
+                    || appImageDir.toFile().list().length == 0) {
```
Comment (shipilev) [inline src/jdk.jpackage/share/classes/jdk/jpackage/internal/DeployParams.java:213]: Are the calls to `list()` idempotent enough here? I.e. can it turn `NULL` on the second call?

**Review by shipilev — APPROVED** [review]
If you are seeking a formal (R)eview ack, here it is.

File: src/jdk.jpackage/share/classes/jdk/jpackage/internal/DeployParams.java (hunk 1 above)
Comment (andyherrick) [reply src/jdk.jpackage/share/classes/jdk/jpackage/internal/DeployParams.java:213]: second call wont be made if first returns null

Comment (shipilev) [reply src/jdk.jpackage/share/classes/jdk/jpackage/internal/DeployParams.java:213]: Not what I was asking. Can first call return *not* `NULL`, and then second return `NULL`, failing with NPE? It is customary to use local variable if we are unsure about this.

File: src/jdk.jpackage/share/classes/jdk/jpackage/internal/DeployParams.java [hunk 2]
```diff
@@ -111,7 +111,7 @@ static void validateName(String s, boolean forApp)
             if (forApp) {
                 return;
             } else {
-                throw new PackagerException(exceptionKey, s);
+                throw new PackagerException(exceptionKey);
```
Comment (azuev-java) [inline src/jdk.jpackage/share/classes/jdk/jpackage/internal/DeployParams.java:114]: Wouldn't it be better to do throw new PackagerException(exceptionKey, "null"); so the message looks more understandable later?

Comment (andyherrick) [reply src/jdk.jpackage/share/classes/jdk/jpackage/internal/DeployParams.java:114]: isn't that exactly the same ?

**Pull request closed without merging**